lib/src/robot_interface/robot_interface.cpp

```diff
@@ -632,15 +671,38 @@ void RobotInterface::setJointCommands(double s0, double s1, double e0, double e1
 
 bool RobotInterface::detectForceInteraction()
 {
-    double f_x = abs(_curr_wrench.force.x - _filt_force[0]);
-    double f_y = abs(_curr_wrench.force.y - _filt_force[1]);
-    double f_z = abs(_curr_wrench.force.z - _filt_force[2]);
+    // ROS_INFO("Filt Forces: %g, %g, %g", _filt_force[0], _filt_force[1], _filt_force[2]);
+    // compare the current force to the filter force. if the difference is above a
+    // threshold defined in utils.h, return true
+
+    double small_thres = exp (-5);
 
-    ROS_DEBUG("Interaction: %g %g %g", f_x, f_y, f_z);
+    std::vector<double> curr_force;
+    curr_force.push_back(_curr_wrench.force.x);
+    curr_force.push_back(_curr_wrench.force.y);
+    curr_force.push_back(_curr_wrench.force.z);
+
+    std::vector<double> curr_diff;
+    curr_diff.push_back(0.0);
+    curr_diff.push_back(0.0);
+    curr_diff.push_back(0.0);
+
+    for (int i = 0; i < 3; ++i)
+    {
+        if (abs(_filt_force[i]) > small_thres)
+        {
+            curr_diff[i] = abs((curr_force[i] - _filt_force[i])/_filt_force[i]);
+        }
+        else
+        {
+            curr_diff[i] = abs((curr_force[i] - _filt_force[i])/(abs(_filt_force[i]) + 0.01));
+        }
+    }
```

Comment:
@omangin in the previous code, `detectForceInteraction` was only checking if the values were above threshold, whereas all the computation was done in `filterForces`. Do you see any reason for having this computation here and not in `filterForces`? I ask you because maybe there is but since I have not a good grasp on how the filter works yet I may be missing something.